compiler/include/astlocs.h

```diff
@@ -41,6 +45,50 @@ class astlocT {
   inline bool operator!=(const astlocT other) const {
     return this->filename != other.filename || this->lineno != other.lineno;
   }
+  inline bool operator<(const astlocT other) const {
+    int strResult = strcmp(this->filename, other.filename);
+    if (strResult == 0) {
+      return this->lineno < other.lineno;
+    }
+    else {
+      return strResult < 0;
+    }
+  }
+  inline bool operator>(const astlocT other) const {
+    int strResult = strcmp(this->filename, other.filename);
+    if (strResult == 0) {
+      return this->lineno > other.lineno;
+    }
+    else {
+      return strResult > 0;
+    }
+  }
+  inline bool operator<=(const astlocT other) const {
+    int strResult = strcmp(this->filename, other.filename);
+    if (strResult == 0) {
+      return this->lineno <= other.lineno;
+    }
+    else {
+      return strResult <= 0;
+    }
+  }
+  inline bool operator>=(const astlocT other) const {
+    int strResult = strcmp(this->filename, other.filename);
+    if (strResult == 0) {
+      return this->lineno >= other.lineno;
+    }
+    else {
+      return strResult >= 0;
+    }
+  }
+
+  inline const char* stringLoc() const {
+    const int tmpBuffSize = 256;
+    char tmpBuff[tmpBuffSize];
+
+    snprintf(tmpBuff, tmpBuffSize, "%s:%d", filename, lineno);
```

Comment:
I know you didn't add it in this PR but this code seems a bit sketchy to me. (Why 256 characters?). Anyway at least AFAIK snprintf always results in a null terminated string even if it overflows.

Reply:
For now, I just added a comment in this function to fix that character count.